src/addressfinder/reverseGeocode.tsx

```diff
@@ -0,0 +1,41 @@
+const addressFinderAPIKey = "ARFHPVK67QXM49BEWDL3";
+const reverseGeocodeURL = `https://api.addressfinder.io/api/nz/address/reverse_geocode/?key=${addressFinderAPIKey}&format=json`;
+
+export interface reverseGeocodeResp {
+  completions: addressResp[];
+  paid: boolean;
+  demo: boolean;
+  success: boolean;
+}
+
+interface addressResp {
+  a: string;
+  pxid: string;
+}
+
+// TODO(2021-09-13): handle error/edge cases better!
+async function getSuburb(lat: number, lng: number): Promise<string> {
+  try {
+    let resp = await fetch(reverseGeocodeURL + `&y=${lat}&x=${lng}`);
```

Comment:
Is there a more elegant way to handle query params with JS/TS fetch?

Reply:
Two ways to handle this:

```javascript
let response = await fetch(`${reverseGeocodeURL}&y=${lat}&x=${lng}`);
```

or:

```javascript
fetch('https://example.com?' + new URLSearchParams({
    foo: 'value',
    bar: 2,
}))
```

Reply:
Yes, the second method was what I was looking for. Ran into issues earlier since `lat`/`lng` are numbers, which doesn't play nicely when casting to `URLSearchParams` which expects string values. Manually casting them as strings work. Updated to use this method.

As an aside, I don't see any difference between the first method (`${reverseGeocodeURL}&y=${lat}&x=${lng}`) and the original code (`reverseGeocodeURL + `&y=${lat}&x=${lng}`), and wouldn't really mind either style as long as it is consistent throughout the codebase.